src/main/java/hunternif/mc/atlas/util/Log.java

```diff
@@ -1,26 +1,22 @@
 package hunternif.mc.atlas.util;
 
-import net.minecraftforge.fml.common.FMLLog;
 import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Logger;
 
 /**
  * A simple logging helper that is not directly dependent on any mod classes.
- * Make sure to call {@link #setModID(String modID)} during the mod's init.
+ * Make sure to call {@link #setModLog(Logger modLog)} during the mod's init.
  * @author Hunternif
  */
 public class Log {
-	private static String modID;
+	private static Logger modLog;
 	
-	public static void setModID(String modID) {
-		Log.modID = modID;
+	public static void setModLog(Logger modLog) {
+		Log.modLog = modLog;
 	}
 	
 	private static void log(Level level, Throwable ex, String msg, Object... data) {
-		if (modID != null) {
-			FMLLog.log(modID, level, ex, msg, data);
-		} else {
-			FMLLog.log(level, ex, msg, data);
-		}
+		modLog.log(level, () -> data == null ? msg : String.format(msg, data), ex);

```

Comment:
modLog might be `null` here

Reply:
That would be bad news, when is logging used before pre-init?